### ISSUE TITLE
Users cannot see their own charts listed in Charts Menu

### ISSUE THREAD

**Issue by helo-ch** [apache/superset#12041]
On SuperSet (0.38.0), all our users have the following roles :
* `Gamma`
* `sql_lab`
* custom role `DB_ACCESS` : `database access on [Timescaledb XXX].(id:3)`
* custom role `tables_modify` : `[can edit on TableModelView, can add on TableModelView, can list on TableModelView, can show on TableModelView, can tables on Superset, refresh on TableModelView, can save on Datasource, can get on Datasource, can datasources on Superset, menu access on Tables, can list on TableColumnInlineView, can show on TableColumnInlineView, can fetch datasource metadata on Superset]`

### Expected results

Users should be able to:
* create charts
* see their charts in Charts Menu, edit them
* add Charts to dashboard
* see their dashboards in Dashboard Menu

### Actual results

Users can:
* create charts
* add Charts to dashboard
* see their dashboards in Dashboard Menu

But they can't see their Charts listed in the Charts Menu. 
They can see them in dashboards and edit them from there, but that's not convenient. 

#### How to reproduce the bug

1. Create user with custom roles as stated earlier
2. Create a Chart with that user, add it to a Dashboard when saving
3. Go to Charts Menu : the chart created is not listed there

### Environment

(please complete the following information):

- superset version: `0.38.0`

### Additional context

I think this issue is the same as mentionned in #6077, #8066 and #8927, but these are all closed with no answer or fix.

Is this the expected behaviour? If yes, what permission should we add to our custom roles so that users can see their own charts? 

#### Update
Additional info : it does work when each `datasource access` permission is added in the corresponding `DB_ACCESS_XXX` role.

**Comment by helo-ch** [comment]
Problem seems to have been solved in version 1.0.1

Users now see their own charts in Charts Menu.

**Comment by helo-ch** [comment]
It appears issue was **not** resolved with the update after all : users can see their old charts and datasets (from before we updated), but they still cannot see the new ones they create.

**Comment by alexryndin** [comment]
This issue has been following me for a very long time. The problem is that superset has a very limited documentation and its access control model is mostly based on FAB which is pretty hard to use by end users. All that remains is to try to add different permission to a role until you get desired result. I've found out that missing permission is schema access on [database].[schema]. After a role is granted that perm, users are able to see created datasources automatically. Have fun!

**Comment by helo-ch** [comment]
> This issue has been following me for a very long time. The problem is that superset has a very limited documentation and its access control model is mostly based on FAB which is pretty hard to use by end users. All that remains is to try to add different permission to a role until you get desired result. I've found out that missing permission is `schema access on [database].[schema]`. After a role is granted that perm, users are able to see created datasources automatically. Have fun!

Thank you, that was it! 
Closing as it is not a bug after all.

**Comment by tobytarczy** [comment]
I have been trying to install apache superset  via python and get to the point where I can login but there are no menus. This is not a great product if it is this cumbersome to install and configure. I would be grateful if someone could point in the direction  where I can sort this out. I am a very advanced Power BI user however I think there is a gap in the market for a truly web based reporting tool Many thanks in advance

**Comment by rusackas** [comment]
@tobytarczy You should be able to spin up Docker following [these instructions](https://superset.apache.org/docs/quickstart) fairly easily. For additional installation help, you can also join [slack](http://bit.ly/join-superset-slack). Alternatively, you can even set up a free account on [Preset](https://preset.io/) if you'd like.

**Comment by tobytarczy** [comment]
Thank you, I have managed to install and configure the product. In all honesty whilst I think this is the right direction it has a ways to go with respect to functionality and ease of use.  I will monitor how the product evolves over time.

Power BI is very good and I have managed to create some amazing dashboards, but it is getting to be too costly and not very good when it comes to displaying on a web page, along side of PowerPages which is cumbersome and too complicated to get something up an running with very little flexibility.

Thank you again for your support and if you have any recommendations I would love to hear them.


Kind Regards,



Toby Tarczy

Calvella

+44 77991 33349



Calvella Limited, reserves the right to monitor e-mails and any replies sent back.  Registered address; Sky View, Argosy Road, East Midlands Airport, Castle Donington, Derby DE74 2SA. Registered in England and Wales, Company No. 6052142; Vat Reg. No 894273877.

________________________________
